app/helpers/proposals_helper.rb

```diff
@@ -0,0 +1,72 @@
+module ProposalsHelper
+  # For the functions relating to the nodes on the progress page, there are
+  # two classes which control the CSS that display the 'graphic' of the progress
+  # The node class controls whether the blue dots are faded (this stage hasn't
+  # been reached), visible (stage has been completed), or has a ring (current).
+  # The line class controls whether the lines are faded and their placement
+  # (since the active ones need to be placed slightly lower).
+
+  # Determine what the first node should look like on the progress page
+  def submit_node
```

Comment:
I extracted the if/elsif/elses from the view to try to clean it up and put them in this helper.  I tried to add comments to explain my logic, but I am unsure if what I wrote is really adding much.